<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>tkBatcherUI</class>
 <widget class="QWidget" name="tkBatcherUI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>724</width>
    <height>840</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>tk Batcher 1.0</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>2</number>
   </property>
   <property name="margin">
    <number>2</number>
   </property>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>File pattern</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="batcherPathLE">
       <property name="text">
        <string>Z:\Toonkit\RnD\Src\GitRepo\oscar\build\OscarEditor\release\Data\Rigs\{NodeName:.+}\{NodeName:.+}.ma</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="batcherRefreshPathBT">
       <property name="text">
        <string>Refresh</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="Nodes">
     <property name="title">
      <string>Nodes</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QListWidget" name="batcherNodesLB">
        <property name="selectionMode">
         <enum>QAbstractItemView::ExtendedSelection</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <property name="topMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QCheckBox" name="batcherSelectionCB">
          <property name="text">
           <string>Batch on selection only</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="batcherOpenSelectedCB">
          <property name="text">
           <string>Open selected</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="batcherExploreSelectedCB">
          <property name="text">
           <string>Explore selected</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Batch</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <property name="topMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>File path :</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="batcherFilePathLE"/>
        </item>
        <item>
         <widget class="QPushButton" name="batcherOpenBT">
          <property name="text">
           <string>Open...</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="batcherSaveBT">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Save...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="topMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Name</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="batcherNameLE">
          <property name="text">
           <string>LogName</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <property name="topMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QPushButton" name="batcherSelectSuccessBT">
          <property name="text">
           <string>Select successfull</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="batcherSelectFailureBT">
          <property name="text">
           <string>Select failed</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="batcherSelectNoneBT">
          <property name="text">
           <string>Select never passed</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTextEdit" name="batcherCodeTE">
        <property name="html">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;#Pre-defined variables&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;#---------------------------------------------------&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;#pathVariables : (dictionary) variables extracted from path pattern&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;#fileName : (string) the file name without extension (read only)&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;#filePath : (string) the full file path (read only)&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;#saveFilePath : (string) the path to save the file to&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;#saveFile : (bool) tells if we have to save the file in the end (default can change, see UI)&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;#success : (bool) tells if the batch is a success (default True, always False if script failed)&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:8pt;&quot;&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;&lt;br /&gt;&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;print fileName&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;saveFile = False&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="batcherCodeLE">
        <property name="visible">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="batcherCodeInputLE">
        <property name="visible">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <property name="topMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QCheckBox" name="batcherSaveByDefaultCB">
          <property name="text">
           <string>Save by default</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Preferred</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Save path</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="batcherSavepathLE">
          <property name="text">
           <string>#filePath</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Preferred</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QCheckBox" name="batcherSkipCB">
          <property name="text">
           <string>Skip if passed</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <property name="topMargin">
         <number>10</number>
        </property>
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Log files output</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="batcherLogSavepathLE">
          <property name="text">
           <string>os.path.join(os.path.split(#filePath)[0], #fileName + &quot;_Batches&quot;)</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <property name="topMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QPushButton" name="batcherSimBT">
          <property name="text">
           <string>Simulate</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="batcherDoBatchBT">
          <property name="text">
           <string>DO BATCH</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="batcherDoMayaBatchBT">
          <property name="text">
           <string>DO WITH MAYABATCH</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_7">
     <property name="topMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QProgressBar" name="batcherProgressBar">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>21</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>15</height>
        </size>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="pixmap">
        <pixmap>../../../../../../../cyril/.designer/backup/pictures/MiniLogo.png</pixmap>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
       <property name="margin">
        <number>0</number>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>batcherDoBatchBT</sender>
   <signal>pressed()</signal>
   <receiver>batcherCodeLE</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>352</y>
    </hint>
    <hint type="destinationlabel">
     <x>101</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batcherDoBatchBT</sender>
   <signal>pressed()</signal>
   <receiver>batcherCodeTE</receiver>
   <slot>selectAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>352</y>
    </hint>
    <hint type="destinationlabel">
     <x>303</x>
     <y>278</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batcherDoBatchBT</sender>
   <signal>pressed()</signal>
   <receiver>batcherCodeTE</receiver>
   <slot>copy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>352</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>285</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batcherDoBatchBT</sender>
   <signal>pressed()</signal>
   <receiver>batcherCodeLE</receiver>
   <slot>paste()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>352</y>
    </hint>
    <hint type="destinationlabel">
     <x>101</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batcherCodeInputLE</sender>
   <signal>textChanged(QString)</signal>
   <receiver>batcherCodeTE</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>130</x>
     <y>311</y>
    </hint>
    <hint type="destinationlabel">
     <x>116</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batcherSimBT</sender>
   <signal>pressed()</signal>
   <receiver>batcherCodeTE</receiver>
   <slot>selectAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>97</x>
     <y>341</y>
    </hint>
    <hint type="destinationlabel">
     <x>184</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batcherSimBT</sender>
   <signal>pressed()</signal>
   <receiver>batcherCodeTE</receiver>
   <slot>copy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>97</x>
     <y>341</y>
    </hint>
    <hint type="destinationlabel">
     <x>184</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batcherSimBT</sender>
   <signal>pressed()</signal>
   <receiver>batcherCodeLE</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>121</x>
     <y>344</y>
    </hint>
    <hint type="destinationlabel">
     <x>101</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batcherSimBT</sender>
   <signal>pressed()</signal>
   <receiver>batcherCodeLE</receiver>
   <slot>paste()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>72</x>
     <y>342</y>
    </hint>
    <hint type="destinationlabel">
     <x>101</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batcherDoMayaBatchBT</sender>
   <signal>pressed()</signal>
   <receiver>batcherCodeLE</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>595</x>
     <y>789</y>
    </hint>
    <hint type="destinationlabel">
     <x>51</x>
     <y>436</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batcherDoMayaBatchBT</sender>
   <signal>pressed()</signal>
   <receiver>batcherCodeTE</receiver>
   <slot>selectAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>595</x>
     <y>789</y>
    </hint>
    <hint type="destinationlabel">
     <x>361</x>
     <y>620</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batcherDoMayaBatchBT</sender>
   <signal>pressed()</signal>
   <receiver>batcherCodeTE</receiver>
   <slot>copy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>595</x>
     <y>789</y>
    </hint>
    <hint type="destinationlabel">
     <x>361</x>
     <y>620</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batcherDoMayaBatchBT</sender>
   <signal>pressed()</signal>
   <receiver>batcherCodeLE</receiver>
   <slot>paste()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>595</x>
     <y>789</y>
    </hint>
    <hint type="destinationlabel">
     <x>51</x>
     <y>436</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
